### PR TITLE
Add stateful object

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by amelentieva — APPROVED** [review]
Cool job!!!

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,27 @@
...
+        for (const keys of action.keysToRemove) {
+          delete state[keys];
+        }
+        break;
+
+      default:
+        for (const keys in state) {
+          delete state[keys];
+        }
+        break;
```
Comment (jstmpelowycz) [inline src/transformState.js:24]:
```suggestion
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;
```

Comment (jstmpelowycz) [reply src/transformState.js:24]: why important: in this task, you're given a clear explanation of the different cases, and here's a slight amount of code, but when it comes to a larger project, other devs may get stuck understanding what's written

Comment (jstmpelowycz) [reply src/transformState.js:24]: also, don't forget to use the single form as you take out only one item, not a bunch

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,27 @@
...
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
```
Comment (jstmpelowycz) [inline src/transformState.js:15]:
```suggestion
        for (const key of action.keysToRemove) {
```

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**